File: ruby/raindrops/raindrops.rb
class Raindrops
  FACTORS = {
    'Pling': 3,
    'Plang': 5,
    'Plong': 7
  }.freeze

  def self.convert(num)
    sounds = FACTORS.map do |sound, factor|
      sound if (num % factor).zero?
    end.join
    sounds.empty? ? num.to_s : sounds
  end
end

File: python/raindrops/raindrops.py
def raindrops(number):
    RAINDROP_TYPES = {
        "Pling": 3,
        "Plang": 5,
        "Plong": 7
    }

    raindrop_str =  "".join([k for k, v in RAINDROP_TYPES.items() if number % v == 0])

    if raindrop_str == "":
        return str(number)
    else:
        return raindrop_str
